HopsanGUI/Dialogs/OptimizationDialog.h

```diff
@@ -76,16 +77,18 @@ public slots:
 private slots:
     void generateScriptSkeleton();
     void openScriptWizard();
-    void saveScriptFile();
```

Comment:
I dont think that the saveScript function taking bool is needed. The one tanking qstring and the one savingAs should be enough.

If the one taking string gets an empty string and mScriptFileInfo is not set (or points to file that does not exist) then the saveAs should be called. 
All file selction code could be contained in that function, then it would call the saveScript function taking qstring with the selected path. 
Later calling saveScript would then save to the existing mScriptFile.

Reply:
But there are two QToolButtons, which each need to be connected to a different slot. We could use a QSignalMapper to provide arguments to the slots, but that would be over-complicated.

It is the same problem in ModelWidget class, which has three functions: save(), saveAs() and saveModel().

I will move all save code to one of the functions though, to make things clearer.

Reply:
Yes two buttons save and save as, then only two slots should be needed. Or maybe I am missing some implementation detail. I have only looked at the PR diff, not the surrounding code.

Reply:
We also use it for saving a backup copy of the script prior to starting the optimization. Therefore we need a save function that can take an arbitrary filepath as a string argument.

Reply:
Yes the one with QString is needed, the one taking bool should not be needed. I think that you can signal the slot taking qstring without argument. Then if the qstring defaults to empty string, that could be used to decide if the saveAs slot should be called. Depends on if the current mSaveFileInfo (or whatever its name is) is set (and what it is set to)

Reply:
The save(qstring) can call saveAs() (if qstring empty and current file empty).
SaveAs opens the dialog and then calls save(qstring=validpath) which saves to the intended file.

Anyway not super important...

Reply:
Ok, I think I managed to solve it with only two functions. Check for current file being empty needs to be done in the save() function before calling saveAs(), but that should not be a problem I think.

Reply:
Have not tested, but looks good now